From 789bf703925c0b83410da857747142850ca0a8fb Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 19 Aug 2015 16:43:25 +0300
Subject: [PATCH 0341/1240] config: added configs for static/dynamic ddr

- added generic configs for static and dynamic code

Change-Id: I849f18778a0ec79a80d554b07214d0113f388dc6
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/22972
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 drivers/ddr/Kconfig     | 40 ++++++++++++++++++++++++++++------------
 drivers/ddr/Makefile    |  6 ++++--
 drivers/ddr/mvebu_ddr.c |  8 +++++---
 3 files changed, 37 insertions(+), 17 deletions(-)

diff --git a/drivers/ddr/Kconfig b/drivers/ddr/Kconfig
index f733d76..fef8441 100644
--- a/drivers/ddr/Kconfig
+++ b/drivers/ddr/Kconfig
@@ -11,32 +11,48 @@ config MVEBU_SPL_DDR_SUPPORT
 	  list below.
 
 choice
-	prompt "Controller select"
+	prompt "Controller static/dynamic select"
 
-config MVEBU_DUNIT_SUPPORT
-	bool "DUNIT controller support"
+config MVEBU_STATIC_SUPPORT
+	bool "Static DDR initialization"
 	depends on MVEBU_SPL_DDR_SUPPORT
 	help
 	  Choose this option to add support
-	  for DUNIT controller.
+	  for static DDR initialization code,
+	  init the DDR with out training.
+	  to add this option to SPL code,
+	  need to add +S:CONFIG_MVEBU_STATIC_SUPPORT to defconfig file
 
-config MVEBU_MK6_STATIC_SUPPORT
-	bool "MK6 static controller support"
+config MVEBU_DYNAMIC_SUPPORT
+	bool "Dynamic DDR initialization"
 	depends on MVEBU_SPL_DDR_SUPPORT
 	help
 	  Choose this option to add support
-	  for static DDR training code for MK6 controller
+	  for dynamic DDR initialization code,
+	  init the DDR with training.
 	  to add this option to SPL code,
-	  need to add +S:CONFIG_MVEBU_MK6_STATIC_SUPPORT to defconfig file
+	  need to add +S:CONFIG_MVEBU_DYNAMIC_SUPPORT to defconfig file
+
+endchoice
+
+choice
+	prompt "Controller select"
+
+config MVEBU_DUNIT_SUPPORT
+	bool "DUNIT controller support"
+	depends on MVEBU_SPL_DDR_SUPPORT
+	help
+	  Choose this option to add support
+	  for DUNIT controller.
 
-config MVEBU_MK6_DYNAMIC_SUPPORT
-	bool "MK6 dynamic controller support"
+config MVEBU_MK6_SUPPORT
+	bool "MK6 controller support"
 	depends on MVEBU_SPL_DDR_SUPPORT
 	help
 	  Choose this option to add support
-	  for dynamic DDR training code for MK6 controller
+	  for DDR code for MK6 controller
 	  to add this option to SPL code,
-	  need to add +S:CONFIG_MVEBU_MK6_DYNAMIC_SUPPORT to defconfig file
+	  need to add +S:CONFIG_MVEBU_MK6_SUPPORT to defconfig file
 
 endchoice
 endmenu
diff --git a/drivers/ddr/Makefile b/drivers/ddr/Makefile
index ae0ff55..86b84c0 100644
--- a/drivers/ddr/Makefile
+++ b/drivers/ddr/Makefile
@@ -1,4 +1,6 @@
 obj-y += mvebu_ddr.o
 obj-$(CONFIG_MVEBU_DUNIT_SUPPORT) += ddr_dunit.o
-obj-$(CONFIG_MVEBU_MK6_STATIC_SUPPORT) += ddr_mckinley6.o
-obj-$(CONFIG_MVEBU_MK6_DYNAMIC_SUPPORT) += mckinley_ddr/
+ifdef CONFIG_MVEBU_MK6_SUPPORT
+obj-$(CONFIG_MVEBU_STATIC_SUPPORT) += ddr_mckinley6.o
+obj-$(CONFIG_MVEBU_DYNAMIC_SUPPORT) += mckinley_ddr/
+endif
diff --git a/drivers/ddr/mvebu_ddr.c b/drivers/ddr/mvebu_ddr.c
index 93320dc..9a248ec 100644
--- a/drivers/ddr/mvebu_ddr.c
+++ b/drivers/ddr/mvebu_ddr.c
@@ -23,7 +23,7 @@
 #include <asm/arch-mvebu/mvebu.h>
 #include <asm/arch-mvebu/fdt.h>
 
-#ifdef CONFIG_MVEBU_MK6_STATIC_SUPPORT
+#ifdef CONFIG_MVEBU_STATIC_SUPPORT
 static void static_dram_init(struct mvebu_dram_config *config)
 {
 	mvebu_dram_mac_init(config);
@@ -65,10 +65,12 @@ void mvebu_dram_init(const void *blob)
 		error("missing DDR PHY base address in DDR PHY node\n");
 		return;
 	}
-#ifdef CONFIG_MVEBU_MK6_STATIC_SUPPORT
+#ifdef CONFIG_MVEBU_STATIC_SUPPORT
 	static_dram_init(config);
-#elif CONFIG_MVEBU_MK6_DYNAMIC_SUPPORT
+#elif CONFIG_MVEBU_DYNAMIC_SUPPORT
 	dynamic_dram_init(config);
+#else
+#error "No DRAM configuration selected"
 #endif
 	debug_exit();
 }
-- 
1.9.1

